core/store/tx_manager.go

```diff
@@ -714,23 +721,61 @@ func (txm *EthTxManager) handleSafe(
 	return nil
 }
 
-// bumpGas creates a new transaction attempt with an increased gas cost
+// BumpGasByIncrement returns a new gas price increased by the larger of either
+// a percentage bump or a fixed size bump
+func (txm *EthTxManager) BumpGasByIncrement(originalGasPrice *big.Int) *big.Int {
+	// Similar logic is used in geth
+	// See: https://github.com/ethereum/go-ethereum/blob/8d7aa9078f8a94c2c10b1d11e04242df0ea91e5b/core/tx_list.go#L255
+	// And: https://github.com/ethereum/go-ethereum/blob/8d7aa9078f8a94c2c10b1d11e04242df0ea91e5b/core/tx_pool.go#L171
+	minimumGasBumpByPercentage := new(big.Int).Div(
+		new(big.Int).Mul(
+			originalGasPrice,
+			big.NewInt(100+int64(txm.config.EthGasBumpPercent())),
+		),
+		big.NewInt(100),
+	)
+	minimumGasBumpByIncrement := new(big.Int).Add(originalGasPrice, txm.config.EthGasBumpWei())
+	if minimumGasBumpByIncrement.Cmp(minimumGasBumpByPercentage) < 0 {
+		return minimumGasBumpByPercentage
+	}
+	return minimumGasBumpByIncrement
+}
+
 func (txm *EthTxManager) bumpGas(tx *models.Tx, attemptIndex int, blockHeight uint64) error {
 	txAttempt := tx.Attempts[attemptIndex]
 
 	originalGasPrice := txAttempt.GasPrice.ToInt()
-	bumpedGasPrice := new(big.Int).Add(originalGasPrice, txm.config.EthGasBumpWei())
 
-	bumpedTxAttempt, err := txm.createAttempt(tx, bumpedGasPrice, blockHeight)
-	if err != nil {
-		return errors.Wrapf(err, "bumpGas from Tx #%s", txAttempt.Hash.Hex())
-	}
+	bumpedGasPrice := txm.BumpGasByIncrement(originalGasPrice)
 
-	logger.Infow(
-		fmt.Sprintf("Tx #%d created with bumped gas %v", attemptIndex+1, bumpedGasPrice),
-		"originalTxHash", txAttempt.Hash,
-		"newTxHash", bumpedTxAttempt.Hash)
-	return nil
+	for {
+		if bumpedGasPrice.Cmp(txm.config.EthMaxGasPriceWei()) > 0 {
```

Comment:
How hard is it to make a high-priority alert which triggers on this failure?

Reply:
I don't know. Don't know anything about alerting. Perhaps one of @thodges-gh @spooktheducks @RyanRHall or @jleeh will know?

Reply:
Yeah, I don't either. It just seems like it should be a major incident, if this happens.

Reply:
I have added prometheus metrics that will allow us to create an alert.

Reply:
This is in a new PR here: https://github.com/smartcontractkit/chainlink/pull/2520

Reply:
:+1:

Reply:
No we aren't to my knowledge, but probably should. Returning an error wouldn't be good enough, it'd have to log an error.